Build with cmake+clang+mingw32-make

when I use `cmake + clang + mingw32-make` to generate `makefiles` and then call `mingw32-make`, then an error come out:
`error: could not open 'error': no such file or directory`
```
[ 89%] Built target SPIRV
[ 91%] Generating E:/Lib/glslang/build/include/glslang/glsl_intrinsic_header.h
[ 92%] Building CXX object StandAlone/CMakeFiles/glslangValidator.dir/StandAlone.cpp.obj
[ 94%] Linking CXX executable glslangValidator.exe
lld-link: warning: ignoring unknown argument '-undefined'
lld-link: error: could not open 'error': no such file or directory
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
StandAlone\CMakeFiles\glslangValidator.dir\build.make:110: recipe for target 'StandAlone/glslangValidator.exe' failed
mingw32-make[2]: *** [StandAlone/glslangValidator.exe] Error 1
CMakeFiles\Makefile2:1155: recipe for target 'StandAlone/CMakeFiles/glslangValidator.dir/all' failed
mingw32-make[1]: *** [StandAlone/CMakeFiles/glslangValidator.dir/all] Error 2
Makefile:144: recipe for target 'all' failed
mingw32-make: *** [all] Error 2
```

- my `cmake` version:
```
C:\Users\g1018>cmake --version
cmake version 3.22.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

- my `clang/clang++` version
```
C:\Users\g1018>clang --version
clang version 13.0.1
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: F:\LLVM\bin

C:\Users\g1018>clang++ --version
clang version 13.0.1
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: F:\LLVM\bin
```

- my `mingw32-make` version:
```
C:\Users\g1018>mingw32-make -v
GNU Make 3.82.90
Built for i686-pc-mingw32
Copyright (C) 1988-2012 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -191,7 +191,11 @@ elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang" AND NOT MSVC)
         # Error if there's symbols that are not found at link time.
         # add_link_options() was added in CMake 3.13 - if using an earlier
         # version don't set this - it should be caught by presubmits anyway.
-        add_link_options("-Wl,-undefined,error")
+        if (WIN32)
+            add_link_options("-Wl,--no-undefined")
+        else()
+            add_link_options("-Wl,-undefined,error")
+        endif()
     endif()
 elseif(MSVC)
     if(NOT ENABLE_RTTI)

